### PR TITLE
[libc] Make add_with_carry and sub_with_borrow constexpr.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/src/__support/math_extras.h [hunk 1]
```diff
@@ -69,48 +67,68 @@ add_with_carry(T a, T b, T carry_in) {
...
 
 template <>
-LIBC_INLINE SumCarry<unsigned char>
+LIBC_INLINE constexpr SumCarry<unsigned char>
 add_with_carry<unsigned char>(unsigned char a, unsigned char b,
                               unsigned char carry_in) {
-  SumCarry<unsigned char> result{0, 0};
-  result.sum = __builtin_addcb(a, b, carry_in, &result.carry);
-  return result;
+  if (__builtin_is_constant_evaluated()) {
```
Comment (michaelrj-google) [inline libc/src/__support/math_extras.h:73]: does this builtin need a guard to check if it's available?

**Review by nickdesaulniers — APPROVED** [review]
Neat, TIL about `__builtin_is_constant_evaluated`.

File: libc/src/__support/math_extras.h (hunk 1 above)
Comment (nickdesaulniers) [reply libc/src/__support/math_extras.h:73]: No: https://godbolt.org/z/r8WY4vh1q

Comment (lntue) [reply libc/src/__support/math_extras.h:73]: I've checked on godbolt, and seems like it works fine on clang 9.0+ and gcc 11+ in all x86-64, arm32, and riscv-32

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Pull request merged**